File: Source/ecs/Source/Library/philipsecommercesdk/src/main/java/com/philips/platform/ecs/microService/callBack/ECSCallback.java
/*
 *  Copyright (c) Koninklijke Philips N.V., 2020
 *
 *  * All rights are reserved. Reproduction or dissemination
 *
 *  * in whole or in part is prohibited without the prior written
 *
 *  * consent of the copyright holder.
 *
 *
 */
package com.philips.platform.ecs.microService.callBack;

/**
 * The interface Iapsdk callback.
 */
public interface ECSCallback<R,E> {
    /**
     * On response.
     *
     * @param result the result
     */
    public void onResponse(R result);

    /**
     * On failure.
     * @param ecsError the error with exception ,error code and message
     */
    public void onFailure(E ecsError);
}

File: Source/ecs/Source/Library/philipsecommercesdk/src/main/java/com/philips/platform/ecs/microService/callBack/ECSCallback.kt
/*
 *  Copyright (c) Koninklijke Philips N.V., 2020
 *
 *  * All rights are reserved. Reproduction or dissemination
 *
 *  * in whole or in part is prohibited without the prior written
 *
 *  * consent of the copyright holder.
 *
 *
 */
package com.philips.platform.ecs.microService.callBack

/**
 * The is a callback interface to give back result to caller .
 */
interface ECSCallback<R, E> {
    /**
     * On response.
     *
     * @param result the result
     */
    fun onResponse(result: R)

    /**
     * On failure.
     * @param ecsError the error with  message (mandatory) , error code (optional) and enum of error type (optional)
     */
    fun onFailure(ecsError: E)
}